alert-common/src/main/java/com/synopsys/integration/alert/common/channel/message/IssueTrackerMessageParser.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.channel.message;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.lang.Nullable;
+
+import com.synopsys.integration.alert.common.channel.model.IssueContentModel;
+import com.synopsys.integration.alert.common.enumeration.ItemOperation;
+import com.synopsys.integration.alert.common.message.model.ComponentItem;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+
+public abstract class IssueTrackerMessageParser extends ChannelMessageParser {
+    public IssueContentModel createIssueContentModel(String providerName, LinkableItem topic, @Nullable LinkableItem subTopic, Set<ComponentItem> componentItems, ComponentItem arbitraryItem) {
+        String title = createTitle(providerName, topic, subTopic, arbitraryItem);
+
+        StringBuilder description = new StringBuilder();
+        description.append(emphasize("Provider: " + providerName));
+        description.append(getLineSeparator());
+        description.append(createLinkableItemString(topic, true));
+        description.append(getLineSeparator());
+        if (null != subTopic) {
+            description.append(createLinkableItemString(subTopic, true));
+            description.append(getLineSeparator());
+        }
+
+        List<String> additionalComments = new ArrayList<>();
+        String additionalDescriptionInfo = createAdditionalDescriptionInfoOrAddToAdditionalComments(description.length(), componentItems, additionalComments);
+        description.append(String.join("", additionalDescriptionInfo));
```

Comment:
Why is there a join here?

Reply:
I'm guessing `createAdditionalDescriptionInfoOrAddToAdditionalComments(...)` returned a collection at one point. This can be removed.